packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,621 @@
+     * See MeshDebugOptions interface for defaults.
+     */
+    private _options: Required<MeshDebugOptions>;
+
+    /**
+     * Material ID color of this plugin instance.
+     * Taken from index `_PluginCount` of `MaterialColors` at time of instantiation.
+     */
+    @serializeAsColor3()
+    private _materialColor: Color3;
+
+    private _isEnabled = true;
+    /**
+     * Whether the mesh debug plugin is enabled in the material.
+     * Defaults to true.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public isEnabled = true;
+
+    private _mode: MeshDebugMode = MeshDebugMode.NONE;
+    /**
+     * Current mesh debug visualization.
+     * Defaults to NONE.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public mode: MeshDebugMode = MeshDebugMode.NONE;
+
+    private _multiply: boolean = true;
+    /**
+     * Whether the mesh debug visualization multiplies with colors underneath.
+     * Defaults to true.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public multiply: boolean = true;
+
+    /** @internal */
+    protected _markAllDefinesAsDirty(): void {
+        this._enable(this._isEnabled);
+        this.markAllDefinesAsDirty();
+    }
+
+    /**
+     * Creates a new MeshDebugPluginMaterial
+     * @param material Material to attach the mesh debug plugin to
+     * @param options Options for the mesh debug plugin
+     */
+    constructor(material: PBRBaseMaterial | StandardMaterial, options?: MeshDebugOptions) {
+        options = options ?? {};
+
+        const defines = new MeshDebugDefines();
+        defines.DBG_MODE = options.mode ?? defines.DBG_MODE;
+        defines.DBG_MULTIPLY = options.multiply ?? defines.DBG_MULTIPLY;
+        super(material, "MeshDebug", 200, defines, true, true);
+
+        const defaults: Required<MeshDebugOptions> = {
+            mode: defines.DBG_MODE,
+            multiply: defines.DBG_MULTIPLY,
+            shadedDiffuseColor: new Color3(1, 1, 1),
+            shadedSpecularColor: new Color3(0.8, 0.8, 0.8),
+            shadedSpecularPower: 10,
+            wireframeThickness: 0.7,
+            wireframeTrianglesColor: new Color3(0, 0, 0),
+            wireframeVerticesColor: new Color3(0.8, 0.8, 0.8),
+            vertexColor: new Color3(0, 0, 0),
+            vertexRadius: 1.2,
+            uvScale: 20,
+            uvPrimaryColor: new Color3(1, 1, 1),
+            uvSecondaryColor: new Color3(0.5, 0.5, 0.5),
+        };
+
+        this._mode = defines.DBG_MODE;
+        this._multiply = defines.DBG_MULTIPLY;
+        this._options = { ...defaults, ...options };
+        this._materialColor = MeshDebugPluginMaterial.MaterialColors[MeshDebugPluginMaterial._PluginCount++ % MeshDebugPluginMaterial.MaterialColors.length];
+
+        if (material.getScene().getEngine().webGLVersion == 1) {
+            Logger.Error("MeshDebugPluginMaterial is not supported on WebGL 1.0.");

```

Comment:
can you force the extension to be disabled in this case so the shader does not break ?